pkg/query/query.go

```diff
@@ -209,34 +227,45 @@ func (s *Statement) Execute() error {
 
 	// Channel for handling of returned maps
 	mapChan := make(chan map[goDB.ExtraKey]goDB.Val, 1024)
-	aggregateChan := make(chan aggregateResult, 1)
-	go aggregate(mapChan, aggregateChan)
+	aggregateChan := aggregate(mapChan)
 
 	// spawn reader processing units and make them work on the individual DB blocks
+	// processing by interface is sequential, e.g. for multi-interface queries
 	for _, workManager := range workManagers {
-		workManager.ExecuteWorkerReadJobs(mapChan)
+		err = workManager.ExecuteWorkerReadJobs(mapChan, memErrors)
+		if err != nil {
+
+			// an error from the routine can only be of type memory error
+			err = errorMemoryBreach
```

Comment:
Since you're not assigning any errors below and can hence only panic, why not just return errorMemoryBreach in l.253?

Reply:
It's mainly so that the `defer` statement works (see l.168). If you have a more elegant way to address that I'd be happy to change it.